app/validators/value_source_validator.py

```diff
@@ -7,16 +9,43 @@ class ValueSourceValidator(Validator):
     SOURCE_REFERENCE_INVALID = "Invalid {} source reference"
     ANSWER_SOURCE_REFERENCE_INVALID = SOURCE_REFERENCE_INVALID.format("answers")
 
+    # Progress source reference errors
+    SOURCE_REFERENCE_CURENT_BLOCK = (
+        "Invalid {} source reference: cannot be current block"
```

Comment:
Are these used for other non progress sources?

Reply:
No only progress sources so far